lib/utils/validateTypes.mjs

```diff
@@ -118,3 +120,45 @@ export function assertString(value) {
 	// eslint-disable-next-line no-console
 	console.assert(isString(value), `"${value}" must be a string`);
 }
+
+/**
+ * @param {unknown} value
+ * @returns {value is (Range['start'] | Range['end'])}
+ */
+export function isPosition(value) {
+	if (!isPlainObject(value)) return false;
+
+	if (!isNumber(value.line)) return false;
+
+	if (!isNumber(value.column)) return false;
+
+	return true;
+}
+
+/**
+ * @param {unknown} value
+ * @returns {value is Range}
+ */
+export function isRange(value) {
```

Comment:
[note] If we wanted to make `isPosition` and `isRange` more strict, we might also need to check that `value` should contain only specific properties (e.g., `line`/`column` or `start`/`end`).

But that may be too strict for now. I think we can leave it as-is.

Reply:
That might also make things harder for rule authors.
The `postcss` `Position` type for example also has an `offset` field.

Making the type too strict would require rule authors to destructure these to ensure that extra fields are erased.

Agree to leave it as-is for now :)

Reply:
Objects are extensible by default in TS.
It's actually really hard to restrict them compared to flow.
i.e. [`{| /* … */ |}`](https://flow.org/en/docs/types/objects/#exact-and-inexact-object-types)
Passing an object with supplementary properties shouldn't be a problem.

ref: microsoft/TypeScript#12936